stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -2250,55 +2250,7 @@ func.func @slice(%arg0: tensor<3x4xi32>) -> tensor<1x2xi32> {
 
```

Comment:
On a related note, how about also having the author of a PR to come up with an explicitly written down testing strategy (in a PR description) similarly to what I wrote in https://github.com/openxla/stablehlo/pull/996#discussion_r1093933011? This way, everyone including future code archaeologists will be able to verify whether tests were skipped on purpose or as oversight.

Reply:
I think it's definitely a good idea that having an explicitly written testing strategy will help see the overall constraints needing tests and also for future viewers which I have updated the PR description as a starter, but I don't think this will scale very well with more complex ops. For example, the PR description is already quite lengthy with 2 inputs and 5 constraints. Taking the largest op as an example, ConvolutionOp has 19 inputs and 28 constraints. This would amount to an enormous burden for authors of such ops, especially when there are parts to the constraints. Wdyt?

Reply:
"This would amount to an enormous burden for authors of such ops, especially when there are parts to the constraints. Wdyt?". I agree that this is a whole lot of work, but otherwise how do we know that we have comprehensively covered the op with tests? I would recommend that we do this, so that we can resolve this question once and for all.

StableHLO is developed as a portability layer between different ML frameworks and ML compilers, and I think that for that goal there is tremendous value in achieving comprehensive coverage. For the complicated ops, this will cost us hours per op (both in author and reviewer time), but I wouldn't be surprised if this saves the community dozens or even hundreds of hours overall.

Reply:
Agreed. This will also save the reviewers time verifying that all constraints are addressed, and future referees of the PR will understand why the tests were written/omitted.